test/event_listeners.test.js

```diff
@@ -0,0 +1,32 @@
+import {
+  addEventToCategoryModal, addEventToNewTaskModal, addSelectUpdaterEvent,
+} from '../src/Controllers/event_listeners';
+
+const ids = ['addCategoryButton', 'newTaskButton', 'middleSectionAddButton', 'upcomingTasks'];
+const [button1, button2, button3] = ids.map(id => {
+  const btn = document.createElement('button');
+  btn.id = id;
+  document.body.appendChild(btn);
+  return btn;
+});
+
+describe('addEventToCategoryModal', () => {
+  it('should add a click event to a button with addCategoryButton Id', () => {
+    addEventToCategoryModal();
+    expect(button1.getAttribute('click-event')).toBe('true');
+  });
+});
+
+describe('addEventToNewTaskModal', () => {
```

Comment:
* [ ] One of the purposes of the concept of single responsibility is the ease of TESTING.
From the code, the `addEventToNewTaskModal` does two things: add an event listener and also does react to the click of that event listener, in the test you are only testing the event listener addition neglecting the other behavior associated with it.
- Now how do we test this, the extreme will be to have a separate method that can be used in different functions to add a specific event listener to the dom element and a separate one that triggers a reaction in response to the click event.
- The second option is to add another test example to capture the second behavior in the same test suite.